src/clj/puppetlabs/services/master/file_serving.clj

```diff
@@ -168,14 +165,13 @@
   modulepath. If the modulepath is not defined or the configuration map passed
   is nil then return the default modulepath."
   [project-config]
-  (let [has-modules-key? (contains? project-config :modules)]
-    (if-let [modulepath (parse-modulepath (get project-config :modulepath))]
-      (if has-modules-key?
-        (concat modulepath [".modules"])
-        modulepath)
-      (if has-modules-key?
-        ["modules" ".modules"]
-        default-project-modulepath))))
+  ;; Note that the :modulepath config key's useful when the project
+  ;; organizes its modules inside relative directories. It doesn't
+  ;; make sense for absolute paths since those may not exist
+  ;; on the Puppetserver host.
+  (-> (get project-config :modulepath "modules")
+      (parse-modulepath)
+      (concat [".modules"])))
```

Comment:
Does it matter that we no longer return a vector?
```
user=> (concat (clojure.string/split "foo" #":") ["foo"])
("foo" "foo")
user=> (reduce conj (clojure.string/split "foo" #":") ["foo"])
["foo" "foo"]
```

Reply:
I don't think so, because Line 174 in the previous code reduced to `(concat (parse-modulepath (get project-config :modulepath)) [".modules"])`, which is symmetric with the code here. So, the previous version had the possibility of returning both a seq and a vector, meaning that the distinction between seq and vector probably did not matter in the first place (as long as the return value was a "Clojure array of strings", which it should be with the new code).

Reply:
Actually, a more simpler summary of ^ is that the new code is Line 174 in the old code, so the return value's still fine.

Reply:
Makes sense.